apps/generic-issuance-client/src/pages/pipeline/PipelineEditSection/PipelineActions.tsx

```diff
@@ -0,0 +1,343 @@
+import { Button, HStack } from "@chakra-ui/react";
+import {
+  BasePipelineOptions,
+  GenericIssuanceSelfResponseValue,
+  PipelineDefinition,
+  PipelineInfoResponseValue
+} from "@pcd/passport-interface";
+import _ from "lodash";
+import React, { ReactNode, useCallback } from "react";
+import styled from "styled-components";
+import {
+  useGIContext,
+  useViewingPipelineDefinition
+} from "../../../helpers/Context";
+import { deletePipeline, savePipeline } from "../../../helpers/Mutations";
+import { useJWT } from "../../../helpers/userHooks";
+import { stringifyAndFormat } from "../../../helpers/util";
+
+export function PipelineActions({
+  user,
+  isAdminView,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  pipelineInfo,
+  pipeline,
+  actionInProgress,
+  setActionInProgress,
+  editorValue,
+  setEditorValue,
+  setEditorMaximized
+}: {
+  actionInProgress;
+  setActionInProgress;
+  user: GenericIssuanceSelfResponseValue;
+  pipeline: PipelineDefinition;
+  pipelineInfo: PipelineInfoResponseValue;
+  isAdminView: boolean;
+  editorValue: string;
+  setEditorValue: React.Dispatch<React.SetStateAction<string>>;
+  setEditorMaximized: React.Dispatch<React.SetStateAction<boolean>>;
+}): ReactNode {
+  const userJWT = useJWT();
+  const ctx = useGIContext();
+  const ownedBySomeoneElse = pipeline.ownerUserId !== user.id;
+  const { historyEntry, pipeline: maybeHistoricPipeline } =
+    useViewingPipelineDefinition(pipeline);
+  const hasEdits = stringifyAndFormat(maybeHistoricPipeline) !== editorValue;
+
+  const onDeleteClick = useCallback(async () => {
+    if (userJWT) {
+      if (!confirm("Are you sure you would like to delete this pipeline?")) {
+        return;
+      }
+      setActionInProgress(`Deleting pipeline '${pipeline.id}'...`);
```

Comment:
Does this need to be set back to `undefined` after the action is complete/failed?